openlibrary/plugins/worksearch/code.py

```diff
@@ -636,16 +636,59 @@ class list_search(delegate.page):
     path = '/search/lists'
 
     def GET(self):
+        i = web.input(q='', offset='0', limit='10')
+
+        lists = self.get_results(i.q, i.offset, i.limit)['response']['docs']
+
+        return render_template('search/lists.tmpl', q=i.q, lists=lists)
+
+    def get_results(self, q, offset=0, limit=100):
         if 'env' not in web.ctx:
             delegate.fakeload()
 
-        i = web.input(q='', offset='0', limit='10')
         keys = web.ctx.site.things({
-            "type": "/type/list", "name~": i.q,
-            "limit": int(i.limit), "offset": int(i.offset)
+            "type": "/type/list", "name~": q,
+            "limit": int(limit), "offset": int(offset)
         })
         lists = web.ctx.site.get_many(keys)
-        return render_template('search/lists.tmpl', q=i.q, lists=lists)
+
+        return {
+            'response': {
+                'docs': lists
+            }
+        }
```

Comment:
Any reason we have the wrapping dictionaries? As opposed to just returning the lists?

Reply:
This was in an attempt to make it closer in return value to the other similar `get_results` methods.  However, as this hits the database and the others access SOLR, it means to do that I needed to wrap it

Can definitely remove that behavior, tho - or if we want to keep it I will add a comment explaining why it was done in this way.

Reply:
Ah, I see what you mean. I think it might be better to remove it for now for better readability. If we want to try to DRY these up further, we can do that in the future :+1: